master/internal/spec_util.go

```diff
@@ -134,3 +139,41 @@ func getTaskSessionToken(ctx context.Context, userModel *model.User) (string, er
 	}
 	return token, nil
 }
+
+func (m *Master) getGenericTaskOnAllocationExit(
```

Comment:
can this just be a function? not a method on the gigantic master struct?

Reply:
Master is used in the returned function. Would there be a better way to retrieve it?

Reply:
i guess i was thinking use the program wide db singleton and then we dont need it

Reply:
So the functions that are used in master actually don't need to be there, I guess the functions is postgres_tasks.go got moved out of it in another PR. So I was able to move this function out as well